pkg/docker/instruction/instruction.go

```diff
@@ -6,26 +6,26 @@ import (
 )
 
 // All supported instruction names
-const (
-	Add         = "add"
-	Arg         = "arg"
-	Cmd         = "cmd"
-	Copy        = "copy"
-	Entrypoint  = "entrypoint"
-	Env         = "env"
-	Expose      = "expose"
-	From        = "from"
-	Healthcheck = "healthcheck"
-	Label       = "label"
-	Maintainer  = "maintainer"
-	Onbuild     = "onbuild"
-	Run         = "run"
-	Shell       = "shell"
-	StopSignal  = "stopsignal"
-	User        = "user"
-	Volume      = "volume"
-	Workdir     = "workdir"
-)
+var DOCKER_INSTRUCTION_NAMES []string = []string{
+	"Add",
+	"Arg",
+	"Cmd",
+	"Copy",
+	"Entrypoint",
+	"Env",
+	"Expose",
+	"From",
+	"Healthcheck",
+	"Label",
+	"Maintainer",
+	"Onbuild",
+	"Run",
+	"Shell",
+	"StopSignal",
+	"User",
+	"Volume",
+	"Workdir",
+}
```

Comment:
**Suggestion:** Consider using a map instead of a slice for DOCKER_INSTRUCTION_NAMES. This will improve the performance of the IsKnown function from O(n) to O(1).
```suggestion
var DOCKER_INSTRUCTION_NAMES map[string]bool = map[string]bool{
	"Add": true,
	"Arg": true,
	"Cmd": true,
	"Copy": true,
	"Entrypoint": true,
	"Env": true,
	"Expose": true,
	"From": true,
	"Healthcheck": true,
	"Label": true,
	"Maintainer": true,
	"Onbuild": true,
	"Run": true,
	"Shell": true,
	"StopSignal": true,
	"User": true,
	"Volume": true,
	"Workdir": true,
}

func IsKnown(name string) bool {
	name = strings.ToLower(name)
	_, ok := DOCKER_INSTRUCTION_NAMES[name]
	return ok
}
```

Reply:
Hi @kcq can I improve the code with above suggestions?